### PR TITLE
[mindUI] Blob HUD refresh

### PR DESCRIPTION
**Pull request by DeityLink**
Alright I'm back home. Let's kick this off with some more conversion of the remaining gui_icons to mindUI. Now only the soulblade's left.

Update: better gauges with graduation and more readable numbers. Also the Health gauge features a smaller victory progress tracker.
imageimageimage
image
image
image

:cl:
* rscadd: The Blob HUD has been refreshed using the brand new UI datum system that we first implemented a few days ago. Beside the back-end changes, the new UI has a few new features. The useless Spawn Normal Blob button has been replaced with a toggle to restrain automatic expansion through walls. The gauge features small symbols that accurately indicate how many points you need to perform certain actions, and the Spawn Blob Core icon now tells you explicitly that you need to place more Nodes to be able to place more Cores when their cost is higher than your point cap. Lastly it also features a new look, with more easily readable numbers and graduations.
* rscadd: When trying to place a special Blob too close from another blob of the same type, the game now tells you exactly how close you are from it.
* rscadd: The right side of the Blob HUD now features two new buttons under the Health gauge. One is a shortcut for removing Blob tiles underneath you. The other opens a small Help panel listing some of the shortcuts. The gauge itself also features a smaller golden gauge on its side that tracks the progress to victory.
* rscadd: Oh also Blob Overminds can now move diagonally.
* imageadd: Special Blobs once again animate when they Pulse. For example Resource blobs emit a puff of purple smoke and Factory blobs play an animation when they spawn a Blob Spore.
* soundadd: The Call Overminds blob power that allows other overminds to jump to your location now plays a sound to notify them more reliably.
* bugfix: Fixed mindless Blob cores not using their cerebrate icon.

### CODE REVIEW

**Comment by DeityLink** [comment]
The maptext numbers on both gauges have been aligned a bit better.

File: __DEFINES/planes+layers.dm [hunk 1]
```diff
@@ -234,6 +235,8 @@ What is the naming convention for planes or layers?
 
 	#define MIND_UI_BACK 				10
 	#define MIND_UI_BUTTON 				11
+	#define MIND_UI_FRONT 				12
+	#define MIND_UI_FRONT 				12
```
Comment (DeityLink) [inline __DEFINES/planes+layers.dm]:
```suggestion
```

File: code/_onclick/mindUI/_mindUI.dm [hunk 2]
```diff
@@ -261,11 +266,17 @@ var/list/mind_ui_ID2type = list()
...
+/datum/mind/proc/ReLoginUIFailsafe()
 	var/mob/M = current
 	if (!current)
 		return
 	if(istype(M.locked_to, /obj/structure/bed/chair/vehicle/adminbus))
 		if (!("Adminbus" in activeUIs))
 			DisplayUI("Adminbus")
+	if(isovermind(M))
+		if (!("Blob" in activeUIs))
+			DisplayUI("Blob")
```
Comment (DeityLink) [inline code/_onclick/mindUI/_mindUI.dm]:
```suggestion
			DisplayUI("Blob")

```

File: code/game/gamemodes/blob/theblob.dm [hunk 3]
```diff
@@ -60,7 +60,7 @@ var/list/blob_overminds = list()
 	time_since_last_pulse = world.time
 
 	if(icon_size == 64)
-		if(!asleep && spawning && !no_morph)
+		if(!asleep && spawning && !no_morph && !istype(src, /obj/effect/blob/core))
```
Comment (Kurfursten) [inline code/game/gamemodes/blob/theblob.dm:63]: this would be better with a var but not big enough deal to hold this up over

istype(src) is ugly though

**Pull request merged**